### PR TITLE
Segment Template

### LINKED ISSUE #513: Segmentation Template For the Model Zoo

### PR DESCRIPTION
**Pull request by ericspod**
Fixes #513 .

### Description
This is the template for segmentation bundles. Users can create their own new bundles by copying and adapting this one. 

Multi-GPU support is a work in progress, it doesn't work for me but it should be fine.

### Status
**Ready**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [ ] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [ ] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [x] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [x] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [x] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [x] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

**Review by wyli — APPROVED** [review]
thanks, it looks good to me

File: models/segmentation_template/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+
+# Template Segmentation Bundle
+
+This bundle is meant to be an example of segmentation in 3D which you can copy and modify to create your own bundle.
```
Comment (yiheng-wang-nv) [inline models/segmentation_template/docs/README.md:4]: A good template. Should we also modify the readme of the repo to mention it, thus users can see it directly?

Comment (ericspod) [reply models/segmentation_template/docs/README.md:4]: I've added a mention in the README, have a look but it's a section we can expand with new templates as we go.

File: models/segmentation_template/configs/test.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  key_val_metric:
+    val_mean_dice:
+      _target_: MeanDice
+      include_background: false
+      output_transform: $monai.handlers.from_engine([@pred, @label])
+  val_handlers: '@handlers'
+
+run:
+- $@evaluator.run()
+- '$print(''Per-image Dice:\n'',@evaluator.state.metric_details[''val_mean_dice''].cpu().numpy())'
```
Comment (yiheng-wang-nv) [inline models/segmentation_template/configs/test.yaml:125]: is it a typo here? I think the comma used here has invalid syntax

Comment (ericspod) [reply models/segmentation_template/configs/test.yaml:125]: I'm pretty sure this is correct, without YAML formatting and bundle sigils it's equivalent to this Python code:

```python
print('Per-image Dice:\n',evaluator.state.metric_details['val_mean_dice'].cpu().numpy())
```

**Comment by yiheng-wang-nv** [comment]
We may also think about how to check if the whole .yaml files do not have typo or syntax issues. It seems we cannot use similar unit tests on this template bundle.
cc @wyli @KumoLiu @Nic-Ma

File: models/segmentation_template/configs/test.yaml (hunk 2 above)
Comment (yiheng-wang-nv) [reply models/segmentation_template/configs/test.yaml:125]: Oh sorry, it's yaml's format. Thanks @ericspod for the explain.

**Comment by ericspod** [comment]
> We may also think about how to check if the whole .yaml files do not have typo or syntax issues. It seems we cannot use similar unit tests on this template bundle.

For this we can simply run any YAML file through `pyyaml`, any JSON file through the in-built `json` library, and then further check any Python expressions by attempting to parse them with the in-built `ast` library. This would do a static check without running the code that should be good enough to catch most things.

**Pull request merged**